### ISSUE TITLE
WireGuard interface failed to come "up"

### ISSUE THREAD

**Issue by r0arkin** [binhex/arch-qbittorrentvpn#65]
Hello!

I am trying to set up WireGuard using Mullvad. I downloaded the configs and put them in the correct directory. I specified the following in the docker compose file:
```
privileged: true
sysctls:
  - net.ipv4.conf.all.src_valid_mark=1
```
The container comes up perfectly fine and then when it goes to create wg0 it fails. Here is the output I'm seeing:
```
2020-11-22 12:13:49,183 DEBG 'start-script' stdout output:
[info] Attempting to bring WireGuard interface 'up'...

2020-11-22 12:13:49,189 DEBG 'start-script' stderr output:
Warning: `/config/wireguard/wg0.conf' is world accessible

2020-11-22 12:13:49,197 DEBG 'start-script' stderr output:
[#] ip link add wg0 type wireguard

2020-11-22 12:13:49,198 DEBG 'start-script' stderr output:
Error: Unknown device type.

2020-11-22 12:13:49,199 DEBG 'start-script' stderr output:
Unable to access interface: Protocol not supported

2020-11-22 12:13:49,200 DEBG 'start-script' stderr output:
[#] ip link delete dev wg0

2020-11-22 12:13:49,201 DEBG 'start-script' stderr output:
Cannot find device "wg0"

2020-11-22 12:13:49,202 DEBG 'start-script' stdout output:
[warn] WireGuard interface failed to come 'up', exit code is '1'
```
Is there something I'm missing? I also set the `ADDITIONAL_PORTS` and `Connection\PortRangeMin=` as the custom port that Mullvad gave me.

Thanks!

Edit: I should also note that OpenVPN works just fine.

**Comment by binhex** [comment]
i would need a full log to debug this:- https://github.com/binhex/documentation/blob/master/docker/faq/help.md

**Comment by r0arkin** [comment]
Here you go!
[log.txt](https://github.com/binhex/arch-qbittorrentvpn/files/5582043/log.txt)

**Comment by binhex** [comment]
ok i see you have some PostUp and PreDown commands in the /config/wireguard/wg0.conf file, so can you remove the following lines from that file:-

```
PostUp = iptables -I OUTPUT ! -o %i -m mark ! --mark $(wg show %i fwmark) -m addrtype ! --dst-type LOCAL -j REJECT && ip6tables -I OUTPUT ! -o %i -m mark ! --mark $(wg show %i fwmark) -m addrtype ! --dst-type LOCAL -j REJECT
PreDown = iptables -D OUTPUT ! -o %i -m mark ! --mark $(wg show %i fwmark) -m addrtype ! --dst-type LOCAL -j REJECT && ip6tables -D OUTPUT ! -o %i -m mark ! --mark $(wg show %i fwmark) -m addrtype ! --dst-type LOCAL -j REJECT
```

**Comment by r0arkin** [comment]
Okay, I removed both of those lines including `PostDown = '/root/wireguarddown.sh'` but the same issue is still happening.

**Comment by binhex** [comment]
ok please can you remove all the other .conf files, this might be muddying the water here, you should have a single .conf file in the folder /config/wireguard/ and it should look similar to this:-

`[Interface]
PrivateKey = <a long key>
Address = <ipv44 address>/32,fc00:bbbb:bbbb:bb01::4:52f/128
DNS = 193.138.218.74

[Peer]
PublicKey = <another long key>
AllowedIPs = 0.0.0.0/0,::0/0
Endpoint = 193.32.249.66:51820`

**Comment by binhex** [comment]
hmm ok i see your log looks a lot cleaner now but you still have the issue, what kernel are you running?, if its older than 5.6 then you cannot use wireguard.

**Comment by r0arkin** [comment]
I've removed all the configs but one and removed `wg0.conf` as well. Then restarted qBit, after which the same error appeared. I stopped the docker, removed the `PostUp` and `PreDown` lines as well. Started qBit again but the same issue is happening. If I look in the wg0.conf, I can see the `PostUp` and `PreDown` lines in there again but the supervisord.log doesn't show that it read them. I'm running docker on CentOS 8:
```
CentOS Linux release 8.2.2004 (Core)
4.18.0-193.28.1.el8_2.x86_64
```
After some more tinkering, here is what fixed it:
```
sudo dnf install epel-release elrepo-release -y
sudo dnf install kmod-wireguard wireguard-tools -y
sudo reboot
```
I tested it by bringing the wg0 interface up myself and it worked. I then started the qBit container and it connected.

Now I'm just playing with getting qBit "connectable". It has the orange connection status. I just added port forwarding to WireGuard on Mullvad's site (and updated `ADDITIONAL_PORTS` and `Connection\PortRangeMin=`) so I'm waiting to see if that resolves it.

It may be worth putting all this in a FAQ if you'd like for other CentOS / Red Hat / Fedora users.
